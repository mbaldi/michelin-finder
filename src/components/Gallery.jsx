import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Spin, Row, Col, Typography, Alert } from 'antd';

import { client } from '../App';
import RestaurantCard from './RestaurantCard';

const { Text } = Typography;

const CardsWrapper = styled.div`
  width: 95%;
`;

const Gallery = ({ location }) => {
  const [restaurants, setRestaurants] = useState([]);
  const [loading, setLoading] = useState(false);

  useEffect(() => {
    const getRestaurants = async () => {
      setLoading(true);
      const { data } = await client.post('/restaurantsForLocation', {
        lat: location.location.lat,
        lng: location.location.lng,
      });
      setRestaurants(data);
      setLoading(false);
    };
    getRestaurants();
  }, [location]);

  return (
    <>
      <Text>Showing results for {location.address}</Text>
      {loading && <Spin size="large" />}
      {restaurants && (
        <CardsWrapper>
          <Row gutter={[32, 32]}>
            {restaurants.map((restaurant) => (
              <Col span={6} key={restaurant.id}>
                <RestaurantCard
                  currentAddress={location.address}
                  restaurant={restaurant}
                />
              </Col>
            ))}
          </Row>
        </CardsWrapper>
      )}
      {!loading && !restaurants.length && (
        <Alert type="info" message="We coulndn't find any restaurants nerby" />
      )}
    </>
  );
};

Gallery.propTypes = {
  location: PropTypes.shape({
    location: PropTypes.shape({
      lat: PropTypes.number,
      lng: PropTypes.number,
    }),
    address: PropTypes.string,
  }).isRequired,
};

Gallery.defaultProps = {
  location: {
    location: {
      lat: null,
      lng: null,
    },
    address: '',
  },
};
export default Gallery;
