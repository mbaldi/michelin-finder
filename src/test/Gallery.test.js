import React from 'react';
import { shallow } from 'enzyme';
import Gallery from '../components/Gallery';

describe('Gallery', () => {
  it('renders component', () => {
    const location = {
      status: 'OK',
      location: {
        lat: 38.8976633,
        lng: -77.0365739,
      },
      address: '1600 Pennsylvania Avenue NW #0, Washington, DC 20500, USA',
    };
    const component = shallow(<Gallery location={location} />);
    expect(component).toMatchSnapshot();
  });
});
