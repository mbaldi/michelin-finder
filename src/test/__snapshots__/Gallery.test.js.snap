// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery renders component 1`] = `
<Fragment>
  <Text>
    Showing results for 
    1600 Pennsylvania Avenue NW #0, Washington, DC 20500, USA
  </Text>
  <styled.div>
    <Row
      gutter={
        Array [
          32,
          32,
        ]
      }
    />
  </styled.div>
  <Alert
    message="We coulndn't find any restaurants nerby"
    type="info"
  />
</Fragment>
`;
